FirebaseAuth/Sources/Backend/RPC/FIRRevokeTokenRequest.m

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import "FirebaseAuth/Sources/Backend/RPC/FIRRevokeTokenRequest.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** @var kRevokeTokenEndpoint
+    @brief The endpoint for the revokeToken request.
+ */
+static NSString *const kRevokeTokenEndpoint =
+    @"revokeToken";  // TODO: Double check the endpoint when backend is ready
+
+/** @var kProviderIDKey
+    @brief The key for the provider that issued the token to revoke.
+ */
+static NSString *const kProviderIDKey = @"providerID";
+
+/** @var kTokenTypeKey
+    @brief The key for the type of the token to revoke.
+ */
+static NSString *const kTokenTypeKey = @"tokenType";
+
+/** @var kTokenKey
+    @brief The key for the token to be revoked.
+ */
+static NSString *const kTokenKey = @"token";
+
+/** @var kIDTokenKey
+    @brief The key for the ID Token associated with this credential.
+ */
+static NSString *const kIDTokenKey = @"idToken";
+
+typedef NS_ENUM(NSInteger, FIRTokenType) {
+  /** Indicates that the token type is unspecified.
+   */
+  FIRTokenTypeUnspecified = 0,
+
+  /** Indicates that the token type is refresh token.
+   */
+  FIRTokenTypeRefreshToken = 1,
+
+  /** Indicates that the token type is access token.
+   */
+  FIRTokenTypeAccessToken = 2,
+
+  /** Indicates that the token type is authorization code.
+   */
+  FIRTokenTypeAuthorizationCode = 3,
+};
+
+@implementation FIRRevokeTokenRequest
+
+- (nullable instancetype)initWithToken:(NSString *)token
+                               idToken:(NSString *)idToken
+                  requestConfiguration:(FIRAuthRequestConfiguration *)requestConfiguration {
+  self = [super initWithEndpoint:kRevokeTokenEndpoint requestConfiguration:requestConfiguration];
+  if (self) {
+    // Apple and Authorization code are the only provider and token type we support for now.
+    _providerID = @"apple.com";
```

Comment:
Since the plan is to support other providers, shouldn't `providerID` be a parameter?

Reply:
The plan is, while our backend only supports Apple for now, so it's limited to Apple to be less error-prone. WDYT?

Reply:
Since it's an internal API, I don't feel strongly either way

Reply:
Added a comment to generalize the initializer in the future. Will keep the api as is.